plugin/pkg/scheduler/algorithm/types_test.go

```diff
@@ -0,0 +1,57 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package algorithm
+
+import (
+	"testing"
+
+	"k8s.io/kubernetes/pkg/api/v1"
+	"k8s.io/kubernetes/plugin/pkg/scheduler/schedulercache"
+)
+
+// EmptyMetadataProducer should returns a no-op MetadataProducer type.
+func TestEmptyMetadataProducer(t *testing.T) {
+	noResources := v1.PodSpec{
+		Containers: []v1.Container{
+			{},
+		},
+	}
+
+	tests := []struct {
+		pod    *v1.Pod
+		labels map[string]string
+	}{
+		{
+			pod: &v1.Pod{Spec: noResources},
+			labels: map[string]string{
+				"foo": "bar",
+			},
+		},
+	}
+
+	for _, test := range tests {
+		nodeNameToInfo := map[string]*schedulercache.NodeInfo{
+			"3": schedulercache.NewNodeInfo(),
```

Comment:
Why need 

```
                pod    *v1.Pod
		pods   []*v1.Pod
```

I think that you can use 
```
pods   []*v1.Pod
```
directly, and set one item for `one pod`, what do you think?